### DISCUSSION TITLE
Wait for a background query to complete before calling getquerydata on the same query.

### DISCUSSION THREAD

**Discussion by sohamad** [TanStack/query#3752] [Q&A]
I have a usecase where i am fetching one query in a tab and then access the cached data in the other tab. But when i move to other tab before actually the data is loaded and added to cache. I get undefined in that case. How can i wait until the data is completely fetched and added to cache and then actually make the getquerydata call.

**Comment by TkDodo** [comment]
> I have a usecase where i am fetching one query in a tab and then access the cached data in the other tab.

how would you do that? The react query cache is in-memory - it cannot be shared between browser tabs (without the experimental broadcast plugin)

Unless you are not talking about browser tabs, then please specify what a `tab` is in your description
